src/Functions/multiIf.cpp

```diff
@@ -414,27 +415,87 @@ class FunctionMultiIf final : public FunctionIfBase
             }
         }
 
-        for (size_t row_i = 0; row_i < rows; ++row_i)
+        if (!execute_by_memcpy)
         {
-            auto & instruction = instructions[inserts[row_i]];
-            size_t index = instruction.source_is_constant ? 0 : row_i;
-            res_data[row_i] = data_cols[inserts[row_i]]->getData()[index];
-            null_map_data[row_i] = null_map_cols[inserts[row_i]] ? null_map_cols[inserts[row_i]]->getData()[index] : 0;
+            for (size_t row_i = 0; row_i < rows; ++row_i)
+            {
+                auto & instruction = instructions[inserts[row_i]];
+                size_t index = instruction.source_is_constant ? 0 : row_i;
+                res_data[row_i] = data_cols[inserts[row_i]]->getData()[index];
+                null_map_data[row_i] = null_map_cols[inserts[row_i]] ? null_map_cols[inserts[row_i]]->getData()[index] : 0;
+            }
+        }
+        else
+        {
+            size_t insert_start_pos = 0;
+            for (size_t row_i = 0; row_i < rows; ++row_i)
+            {
+                S curr_insert = inserts[row_i];
+                if (row_i != rows -1 && curr_insert == inserts[row_i + 1])
+                {
+                    continue;
+                }
+                else
+                {
+                    if (instructions[curr_insert].source_is_constant)
+                    {
+                        for (size_t i = insert_start_pos; i <= row_i; ++i)
+                        {
+                            res_data[i] = data_cols[curr_insert]->getData()[0];
+                            null_map_data[i] = null_map_cols[curr_insert]->getData()[0];
+                        }
+                    }
+                    else
+                    {
+                        memcpy(res_data.data() + insert_start_pos,
+                            data_cols[curr_insert]->getData().data() + insert_start_pos, sizeof(T) * (row_i + 1 - insert_start_pos));
+                        memcpy(null_map_data.data() + insert_start_pos,
+                            null_map_cols[curr_insert]->getData().data() + insert_start_pos, sizeof(UInt8) * (row_i + 1 - insert_start_pos));
+                    }
+                    insert_start_pos = row_i + 1;
+                }
+            }
         }
     }
 
     template <typename T, typename S>
-    static void executeInstructionsColumnar(std::vector<Instruction> & instructions, size_t rows, const MutableColumnPtr & res)
+    static void executeInstructionsColumnar(std::vector<Instruction> & instructions, size_t rows, const MutableColumnPtr & res, bool execute_by_memcpy)
```

Comment:
Need to prove that execute_by_memcpy = true has beffer performance than execute_by_memcpy  = false by adding a perf test.

Reply:
On the pr decription